tools/SampleAnalyzer/Interfaces/HEPTopTagger/HTT.h

```diff
@@ -0,0 +1,168 @@
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with MadAnalysis 5. If not, see <http://www.gnu.org/licenses/>
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef MADANALYSIS5_HTT_H
+#define MADANALYSIS5_HTT_H
+
+// SampleAnalyser headers
+#include "SampleAnalyzer/Interfaces/substructure/Commons.h"
+
+namespace fastjet {
+    namespace HEPTopTagger {
+        class HEPTopTagger;
+    }
+}
+
+namespace MA5 {
+    namespace Substructure {
+        class HTT {
+
+        protected:
+            fastjet::HEPTopTagger::HEPTopTagger* _tagger;
+
+        public:
+
+            enum Mode {
+                EARLY_MASSRATIO_SORT_MASS,     // apply 2D mass plane requirements then select the candidate which minimizes |m_cand-mt|
+                LATE_MASSRATIO_SORT_MASS,      // select the candidate which minimizes |m_cand-mt|
+                EARLY_MASSRATIO_SORT_MODDJADE, // apply the 2D mass plane requirements then select the candidate with highest jade distance
+                LATE_MASSRATIO_SORT_MODDJADE,  //select the candidate with highest modified jade distance
+                TWO_STEP_FILTER                // only analyze the candidate built with the highest pT(t) after unclustering
+            };
+
+            struct InputParameters {
+                Mode mode = EARLY_MASSRATIO_SORT_MASS; // execution mode
+
+                MAbool do_optimalR = true; // initialize optimal R or set to fixed R
+                // optimal R parameters
+                MAfloat32 optimalR_min = 0.5; // min jet size
+                MAfloat32 optimalR_step = 0.1; // step size
+                MAfloat32 optimalR_threshold = 0.2; // step size
+
+                // massdrop - unclustering
+                MAfloat32 mass_drop = 0.8;
+                MAfloat32 max_subjet = 30.; // set_max_subjet_mass
+
+                // filtering
+                MAuint32 filt_N = 5; // set_nfilt
+                MAfloat32 filtering_R = 0.3; // max subjet distance for filtering
+                MAfloat32 filtering_minpt = 0.; // min subjet pt for filtering
+                // jet algorithm for filtering
+                Algorithm filtering_algorithm = Algorithm::cambridge;
+
+                // Reclustering
+                // reclustering jet algorithm
+                Algorithm reclustering_algorithm = Algorithm::cambridge;
+
+                //top mass range
+                MAfloat32 top_mass = 172.3;
+                MAfloat32 W_mass = 80.4;
+                MAfloat32 Mtop_min = 150.;
+                MAfloat32 Mtop_max = 200.; //set_top_range(min,max)
+
+                // set top mass ratio range
+                MAfloat32 fw = 0.15;
+                MAfloat32 mass_ratio_range_min = (1-fw)*80.379/172.9;
```

Comment:
We need a function that update this when the user change the mass of the top or the W boson.